x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/cohere/CohereStreamingProcessor.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.external.cohere;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.ElasticsearchStatusException;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.xcontent.XContentFactory;
+import org.elasticsearch.xcontent.XContentParser;
+import org.elasticsearch.xcontent.XContentParserConfiguration;
+import org.elasticsearch.xcontent.XContentType;
+import org.elasticsearch.xpack.core.inference.results.StreamingChatCompletionResults;
+import org.elasticsearch.xpack.inference.common.DelegatingProcessor;
+
+import java.io.IOException;
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.Map;
+import java.util.Optional;
+
+class CohereStreamingProcessor extends DelegatingProcessor<Deque<String>, StreamingChatCompletionResults.Results> {
+    private static final Logger log = LogManager.getLogger(CohereStreamingProcessor.class);
+
+    @Override
+    protected void next(Deque<String> item) throws Exception {
+        if (item.isEmpty()) {
+            // discard empty result and go to the next
+            upstream().request(1);
+            return;
+        }
+
+        var results = new ArrayDeque<StreamingChatCompletionResults.Result>(item.size());
+        for (String json : item) {
+            try (var jsonParser = jsonParser(json)) {
+                var responseMap = jsonParser.map();
+                var eventType = (String) responseMap.get("event_type");
+                switch (eventType) {
+                    case "text-generation" -> parseText(responseMap).ifPresent(results::offer);
+                    case "stream-end" -> validateResponse(responseMap);
+                    case "stream-start", "search-queries-generation", "search-results", "citation-generation", "tool-calls-generation",
+                        "tool-calls-chunk" -> {
+                        // skip
```

Comment:
Should we log if this happens or anything?

Reply:
We should add a debug log - these are included in most responses, and I'm not sure what to do with them or if they're even needed in our response object (for now